src/modules/claim/claim.service.ts

```diff
@@ -149,6 +149,15 @@ export class ClaimService {
     return this.claimRepository.findOne(id);
   }
 
+  /**
+   * returns claims requested for given DIDs
+   * @param dids claim subjects DIDs
+   */
+  public async getBySubjects(dids: string[]): Promise<Claim[]> {
+    const claims = await this.claimRepository.find();
+    return claims.filter((claim) => dids.includes(claim.subject));
```

Comment:
The problem is that subject isn't stored in database, but is encoded in token

Reply:
Ah yes, I see that now. How hard to do you think it would be to store the subject in the database? I notice that the subject is already in the entity, it just isn't stored as a field. Do you know why that is the case?https://github.com/energywebfoundation/iam-cache-server/blob/develop/src/modules/claim/claim.entity.ts